src/server/events/index.ts

```diff
@@ -97,12 +91,13 @@ export async function addOrUpdateEvent(
 		description: eventInput.description,
 		location: eventInput.location,
 		eventType: eventInput.eventType,
+		gcalID: eventInput.gcalID,
 	} as EventDbObject;
-
 	const { value, lastErrorObject, ok } = await models.Events.findOneAndUpdate(
-		{ name: eventDiffObj.name },
+		eventDiffObj.gcalID ? { gcalID: eventDiffObj.gcalID } : { _id: eventDiffObj._id },
```

Comment:
so basically, if the input gives a gcalID, you find by gcalID, otherwise, you find by mongoDB id?

Reply:
why would an input not have a gcal id?

Reply:
Yes, an input may not have a gcal id if you're trying to construct an event without pulling from gcal i.e. some events table UI; constructing obscure internal events not for public use